### ISSUE TITLE
system tables for inner metrics

### ISSUE THREAD

**Issue by MichaelScofield** [GreptimeTeam/greptimedb#602]
### What problem does the new feature solve?

It's good to have a metrics table like the one in [Databend](https://databend.rs/doc/sql-reference/system-tables/system-metrics) or [ClickHouse](https://clickhouse.com/docs/en/operations/system-tables/metrics/). Users can easily "select" on the metrics table to see db's important running status. With metrics table, we can integrate other visualization tools or dashboards that supports SQL like Grafana, to export some meaningful timeseries data.

### What does the feature do?

Adds system tables for inner metrics.

### Implementation challenges

1. In datanode, create a new table called `metrics`, with schema like the [`MetricsTable`](https://github.com/datafuselabs/databend/blob/main/src/query/storages/system/src/metrics_table.rs) in Databend. Register it into default catalog like our `ScriptsTable` do.
2. Implement "select" for metrics table. Instead of actually storing the metrics, call the `render()` method of `PROMETHEUS_HANDLE`, and parse the result (prometheus lines) as table ouput. Refer to the code [here](https://github.com/datafuselabs/databend/blob/main/src/query/storages/system/src/metrics_table.rs#L43) in Databend for details.
3. In frontend, create the same metrics table as well. Unlike `DistTable`, the metrics table does not do distributed query.

**Comment by azhsmesos** [comment]
Can you describe it in more detail？

**Comment by killme2008** [comment]
@azhsmesos  Looks like the metrics table in clickhouse

https://clickhouse.com/docs/en/operations/system-tables/metrics/

**Comment by waynexia** [comment]
This issue requests one or more table in system catalog that contains information about running status. E.g., load, how much memory is used, how many running queries, how many opened fds, etc. It exposes a convenient way to do system analysis and debugging

**Comment by Lilit0x** [comment]
I don't know if this has been implemented since I can't find any indication that it has been, but I find it interesting, and I'd like to work on it. I still have some pending PRs, so this is just like an indication of my interest. 
I'll come back to outline my thought process for it coupled with the implementation details that have been described above. 
Thanks

**Comment by MichaelScofield** [comment]
@Lilit0x Are you still interested on this issue?

**Comment by Lilit0x** [comment]
Hi @MichaelScofield , yes, I am, but I just got assigned to another issue.
If it isn't urgent, I'll like to come back to it.

**Comment by MichaelScofield** [comment]
@Lilit0x Not urgent. Just a friendly ping. Come back to this issue when you have time, thx~

**Comment by waynexia** [comment]
> Hi @MichaelScofield , yes, I am, but I just got assigned to another issue. If it isn't urgent, I'll like to come back to it.

Thanks for your enthusiasm! This is not hurry, and we are making some changes related to our internal metrics https://github.com/GreptimeTeam/greptimedb/issues/1906. Will update this issue when it's unblocked :heart:

**Comment by killme2008** [comment]
> @waynexia @killme2008 I think some of the features needed by this issue have been implemented in information_schema.
> 
> Perhaps we can close this issue and open a new issue if there is any ongoing items.

Yes. We have `runtime_metrics` and try to implement `xxx_stats` etc. in `information_schema`.
